src/utils/dom.ts

```diff
@@ -92,7 +92,14 @@ export function selectorToElement(selector: string): HTMLElement {
   const [tag, attributes] = parseSimpleCssSelector(selector)
   const element = document.createElement(tag ?? 'div')
   for (const name of Object.keys(attributes)) {
-    element.setAttribute(name, attributes[name].join(' '))
+    const value = attributes[name].join(' ')
+    // Changing the `style` attribute can cause a CSP error, therefore we change the `style.cssText` property.
+    // https://github.com/fingerprintjs/fingerprintjs/issues/733
+    if (name === 'style') {
+      element.style.cssText = value
```

Comment:
I think the safest option here (regarding CSP) would be to parse the value and then use `.style.<specific-property> = 'single value'`. The reason is that `.cssText` requires the browser to parse the provided string and [according to the spec this should be gated by `unsafe-eval`](https://www.w3.org/TR/CSP3/#directive-style-src). Also MDN lists this operation as [unsafe](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src#unsafe_style_expressions).

Regardless, I tested it in latest Firefox and Chrome, and `HTMLElement.style.cssText` doesn't seem to be blocked if `unsafe-eval` is missing, which might be explained by [this w3c comment](https://csplite.com/csp145/#w3c_note):

> HTMLElement.style (eg el.style.color='green';) should not be blocked in the absence of 'unsafe-eval' nor in the absence of 'unsafe-inline'

but: 

> .cssText, .insertRule() are meant to block in the absence of 'unsafe-eval'.

So I think we have two options:
1. Keep using `.cssText` and get back to this if / when browsers change their behaviour to match the specification (if I'm reading it correctly).
2. Parse the selectors and use the specific style property with a single value (this might be too difficult to do reliably, but covering only the few `style` occurrences that we currently have in the rules shouldn't be too hard I think).

wdyt?

Reply:
Thank you for the findings. I've also found that the Chromium team [tried to implement](https://bugs.chromium.org/p/chromium/issues/detail?id=336413) the `unsafe-eval` restriction, but they rolled back the changes because it broke many websites.

Even though the `unsafe-eval` policy enablement is very unlikely, I've implemented a solution that doesn't use `cssText` to reduce the risk.